src/pytorch_lightning/tuner/lr_finder.py

```diff
@@ -38,6 +39,10 @@
 else:
     from tqdm import tqdm
 
+_MATPLOTLIB_AVAILABLE: bool = _RequirementAvailable("matplotlib")  # type: ignore[assignment]
```

Comment:
There are two things that come to mind - either explicitly cast it to `bool` somehow, or make all the typing for things like `_*_AVAILABLE` of type `_RequirementAvailable`, which can be used in a boolean comparison because of overwritten `__bool__`

Reply:
Oh nice. There is also a similar annotation in another file, as mentioned by @akihironitta:

https://github.com/Lightning-AI/lightning/blob/cf189bdc9b32326ea02b9af6dd6e1bf04f0e4d2e/src/pytorch_lightning/strategies/deepspeed.py#L56-L58

Should we remove the annotation as well?

Reply:
Yes, please

Reply:
Thanks!